lib/ex_doc/formatter/html/templates.ex

```diff
@@ -74,10 +74,14 @@ defmodule ExDoc.Formatter.HTML.Templates do
   def synopsis(nil), do: nil
 
   def synopsis(doc) when is_binary(doc) do
-    case :binary.split(doc, "</p>") do
-      [left, _] -> String.trim_trailing(left, ":") <> "</p>"
-      [all] -> all
-    end
+    doc =
+      case :binary.split(doc, "</p>") do
+        [left, _] -> String.trim_trailing(left, ":") <> "</p>"
+        [all] -> all
+      end
+
+    ## Remove any anchors found in synopsis.
+    Regex.replace(~r|(<[^>]*) id="[^"]*"([^>]*>)|, doc, ~S"\1\2", [])
```

Comment:
Anchors in synopsis should be fine, no? If they are shown buggy, then we fix that instead?

Reply:
Since the synopsis is present in two places, the anchor will be duplicated. I think that when setting an anchor in the docs the user want it to be in the main body of the documention and not at the listing of all functions.

Reply:
Oh, I get it now. Follow up question: what are you doing that you end-up with anchors in synopsis? :D

Reply:
We have (lots of) old function docs where an anchor has been put at the beginning of the docs in order to link to it using a name. This was from before the time when we added the possibility to link directly to `M:F/A` in the xml docs. So when converting those old docs to markdown, we endup with anchors in the synopsis.

Ideally I would like to remove them from our docs completely, but there are enough of them that it is a lot of work to fix so I did this change instead :)

Reply:
Cool, I pushed a code suggestion.